src/wasm/wat-parser.cpp

```diff
@@ -562,6 +586,22 @@ template<typename Ctx> struct TypeParserCtx {
     }
     return it->second;
   }
+
+  std::vector<char> makeDataString() { return {}; }
+  void appendDataString(std::vector<char>& data, std::string_view str) {
+    data.insert(data.end(), str.begin(), str.end());
+  }
+
+  Limits makeLimits(uint64_t n, std::optional<uint64_t> m) {
+    return m ? Limits{n, *m} : Limits{n, Memory::kUnlimitedSize};
+  }
+  Limits getLimitsFromData(const std::vector<char>& data) {
+    uint64_t size = (data.size() + Memory::kPageSize - 1) / Memory::kPageSize;
+    return {size, size};
+  }
+
+  MemType makeMemType32(Limits limits) { return {Type::i32, limits}; }
+  MemType makeMemType64(Limits limits) { return {Type::i64, limits}; }
```

Comment:
Perhaps a single function that takes the type as a parameter? I don't feel strongly though.

Reply:
I don't want the core parser templates to have to create `Type`s (since they should ideally be reusable outside Binaryen), so the only alternative would be to use a `bool` or a new enum. But these functions are so simple that just having two functions seems simpler and more readable than either of those options.